Userland/Libraries/LibJS/Runtime/DateConstructor.cpp

```diff
@@ -161,14 +161,24 @@ static double parse_date_string(ByteString const& date_string)
     //        Both Chrome and Firefox seem to support "4/17/2019 11:08 PM +0000" with most parts
     //        being optional, however this is not clearly documented anywhere.
     static constexpr auto extra_formats = AK::Array {
-        "%a %b %d %Y %T GMT%z (%+)"sv, // "Tue Nov 07 2023 10:05:55 GMT-0500 (Eastern Standard Time)"
-        "%a, %d %b %Y %T %Z"sv,        // "Tue, 07 Nov 2023 15:05:55 GMT"
-        "%a %b %e %T %z %Y"sv,         // "Wed Apr 17 23:08:53 +0000 2019"
-        "%m/%e/%Y"sv,                  // "4/17/2019"
-        "%m/%e/%Y %R %z"sv,            // "12/05/2022 10:00 -0800"
-        "%Y/%m/%e %R"sv,               // "2014/11/14 13:05"
-        "%Y-%m-%e %R"sv,               // "2014-11-14 13:05"
-        "%B %e, %Y %T"sv,              //  "June 5, 2023 17:00:00"
+        "%a %b %d %Y %T GMT%z (%+)"sv,   // "Tue Nov 07 2023 10:05:55 GMT-0500 (Eastern Standard Time)"
+        "%a, %d %b %Y %T %Z"sv,          // "Tue, 07 Nov 2023 15:05:55 GMT"
+        "%a %b %e %T %z %Y"sv,           // "Wed Apr 17 23:08:53 +0000 2019"
+        "%m/%e/%Y"sv,                    // "4/17/2019"
+        "%m/%e/%Y %R %z"sv,              // "12/05/2022 10:00 -0800"
+        "%Y/%m/%e %R"sv,                 // "2014/11/14 13:05"
+        "%Y-%m-%e %R"sv,                 // "2014-11-14 13:05"
+        "%B %e, %Y %T"sv,                // "June 5, 2023 17:00:00"
+        "%b %d %Y %Z"sv,                 // "Jan 01 1970 GMT"
+        "%a %b %e %T %Y %z"sv,           // "Wed Apr 17 23:08:53 2019 +0000"
+        "%Y-%m-%e %RZ"sv,               // "2021-07-01 03:00Z"
+        "%a, %e %b %Y %T %z"sv,         // "Wed, 17 Jan 2024 11:36:34 +0000"
+        "%a %b %e %Y %T GMT %x (%+)"sv, // "Sun Jan 21 2024 21:11:31 GMT 0100 (Central European Standard Time)"
+        "%Y-%m-%e %T"sv,                // "2024-01-15 00:00:01"
+        "%a %b %e %Y %T  %Z"sv,         // "Tue Nov 07 2023 10:05:55  UTC"
+        "%a %b %e %T %Y"sv,             // "Wed Apr 17 23:08:53 2019"
+        "%Y-%m-%eT%T.%3+%z"sv,          // "2024-01-26T22:10:11.306+0000" this is broken
```

Comment:
I am not sure why this is broken. I made sure that the "%3+" format token made the lexer point at the correct character (it points at the +), but "%z" is failing to yield a correct datetime.

Reply:
You need to increment `format_pos` if you're going to consume an additional character from the format string:
```c++
            if (format_pos + 1 < format.length() && format[format_pos + 1] == '+') {
                auto to_skip = format[format_pos] - '0';
                dbgln("skipping {}", to_skip);
                auto discarded = string_lexer.consume(to_skip);
                dbgln("discarded {}. now pointing at {}", discarded,string_lexer.peek());

				++format_pos; // This fixes the parser
            }
```

However this is the only format specifier that has more than one character in it, so maybe let's avoid that oddity and have the specifier just be "%3" instead of "%3+"?

Alternatively, add a millisecond format specifier (which we drop after parsing, since `struct tm` doesn't support sub-second times). But it can at least check if we parsed something like a decimal followed by 3 digits.